nix/elasticsearch.nix

```diff
@@ -0,0 +1,196 @@
+# Based on https://github.com/cachix/devenv/blob/main/src/modules/services/elasticsearch.nix
+# Note: 
+#   To use elastic search you will need the following inside your `perSystem`, if not already configured:
+#   { 
+#     _module.args.pkgs = import inputs.nixpkgs {
+#        inherit system;
+#        # Required for elastic search
+#        config.allowUnfree = true;
+#      };
+#   }
+{ pkgs, lib, name, config, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options = {
+    enable = lib.mkEnableOption name;
+
+    package = lib.mkPackageOption pkgs "elasticsearch7" { };
+
+    dataDir = lib.mkOption {
+      type = types.str;
+      # TODO: elasticsearch requires absolute path, is there a way to avoid this?
+      default = "$PWD/data/${name}";
```

Comment:
I am not sure if this is the right way to go about this, but it works.

Reply:
Why not compute absolute path in the script?

```bash
 export ES_HOME="realpath ${config.dataDir}"
```